### PR TITLE
fix: Expiring Activity

### PR DESCRIPTION
**Pull request by Muuushuuu**
feature branch https://github.com/Infomaniak/ios-kDrive/pull/1925

Depends On #https://github.com/Infomaniak/ios-core/pull/201

### CODE REVIEW

**Review by adrien-coye — APPROVED** [review]
LGTM 👍

**Review by adrien-coye — CHANGES_REQUESTED** [review]
Discussed, some copilot notes to check

File: kDriveCore/Data/Upload/UploadQueue/Queue/UploadQueue+Queue.swift [hunk 1]
```diff
@@ -48,10 +48,14 @@ extension UploadQueue: UploadQueueable {
...
 
     public func waitForCompletionIsActive(_ completionHandler: @escaping () -> Void) {
-        Task { [weak self] in
-            guard let self else { completionHandler(); return }
-            while self.isActive {
-                try? await Task.sleep(for: .milliseconds(200))
+        guard isActive else {
+            completionHandler()
+            return
+        }
```
[the remark this answers is not in the document]
Comment (Muuushuuu) [reply kDriveCore/Data/Upload/UploadQueue/Queue/UploadQueue+Queue.swift:54]: Fix this with a guard in `withCheckedContinuation` + `onSuspendedHandler`

File: kDriveCore/Data/Upload/UploadQueue/Queue/UploadQueue.swift [hunk 2]
```diff
@@ -117,3 +124,46 @@ public class UploadQueue: ParallelismHeuristicDelegate {
...
 }
+
+extension UploadQueue: UploadQueueDelegate {
+    public func operationQueueBecameEmpty() {
+        expiringActivity?.endAll()
+        expiringActivity = nil
+        onEmptyHandler?()
+        onEmptyHandler = nil
+        externalDelegate?.operationQueueBecameEmpty()
+    }
```
[the remark this answers is not in the document]
Comment (Muuushuuu) [reply kDriveCore/Data/Upload/UploadQueue/Queue/UploadQueue.swift:137]: @adrien-coye Can we take a look at this together?

File: kDriveCore/Data/Upload/UploadQueue/Queue/UploadQueue.swift [hunk 3]
```diff
@@ -39,6 +39,11 @@ public class UploadQueue: ParallelismHeuristicDelegate {
 
     public var fileUploadedCount = 0
     public var fileUploadFailedCount = 0
+
+    private var expiringActivity: ExpiringActivity?
+
+    var onEmptyHandler: (() -> Void)?
+
```
[the remark this answers is not in the document]
Comment (Muuushuuu) [reply kDriveCore/Data/Upload/UploadQueue/Queue/UploadQueue.swift:47]: @adrien-coye Can we take a look at this together?

**Comment by adrien-coye** [comment]
I did test on iOS16 with an iPhone 8, all works transparently 👏

File: kDriveCore/Data/Upload/UploadQueue/Queue/UploadQueue.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (adrien-coye) [reply kDriveCore/Data/Upload/UploadQueue/Queue/UploadQueue.swift:47]: I did check on my side this comment is no longer valid. All good 👍

File: kDriveCore/Data/Upload/UploadQueue/Queue/UploadQueue.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (adrien-coye) [reply kDriveCore/Data/Upload/UploadQueue/Queue/UploadQueue.swift:137]: I also did check on my side this comment is no longer valid. All good @Muuushuuu  👍

**Review by adrien-coye — APPROVED** [review]
I took the time to test this PR in detail on both a device with iOS 26 and iOS 16
The behaviour on iOS 16 is unchanged and the behaviour on 26 is working well so far.
I noticed some edge cases regarding photo sync I'll discuss with the TL later on.
This is good work, and we can start a real QA pass 🎉
